clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/TopicPartitionReplicaRole.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.davinci.kafka.consumer;
+
+import com.linkedin.venice.pubsub.api.PubSubTopic;
+import com.linkedin.venice.pubsub.api.PubSubTopicPartition;
+
+
+/**
+ * This class is for passing the information regarding the role of the partition replica on that host to
+ *  {@link AggKafkaConsumerService} to achieve finer granularity of consumer assignment. Those information should be
+ *  triggered by helix state transition and properly managed by {@link StoreIngestionTask} to be sent to
+ *  {@link AggKafkaConsumerService}. Currently, the role of future/current/backup for version and leader/follower for
+ *  partition are supported. We could add more information regarding this partition replica if needed in the future.
+ */
+
+public class TopicPartitionReplicaRole {
```

Comment:
Overall it looks good, but I am thinking an alternative way to implement this info. I see you create this class as a more stateful representation of Replica, but when you are using this, you only use VersionRole alone and don't use other info wrapped inside, like you still check leader or not using PCS info, instead of isLeaderReplica() API you created. 

Do you think it would be better either to make this class stronger representation via making this role settable and you always use this one object throughout the lifecycle of a Replica? or maybe even just simplify this to only define a VersionRole and pass it and use it as you are doing right now, instead of wrapping with other info?

Reply:
I agree with Jialin - we should make this as a proper Replica abstraction or just trim it down to bare minimum required fields.

Reply:
So first, this class is not for stateful representation of replica, but for a simplified interface for bottem `KafkaConsumerService` by passing information of topic partition replica, isLeader, isCurrentVersion. So we could include these 3 information in one class, or we could just pass 3 into the interface. And it is possible that in the future we would have more information to let `KafkaConsumerService`  know. Regarding stateful replica abstraction, I think now PCS already played that role.

Reply:
Shall we change it to something called `TopicPartitionInfo` or `TopicPartitionIngestionContext` to more a little generic as we may add more information later?
From the usage, it is certainly not a stateful object, but rather a collection of params passing to `KCS`.

Reply:
No I mean in this case, the only thing you are passing into is the VersionRole. What I meant is if you don't need to keep it as stateful object that you can maintain the state for a Replica abstraction (which I think we eventually may/should), then you can just keep the VersionRole Enum as other information in this class you created are not using it in the code.... It is certainly not wrong, just I think it can be simplified. We can come back to this comment later.